### DISCUSSION TITLE
Best way to deal with line breaks in VARCHAR type?

### DISCUSSION THREAD

**Discussion by Donny-H** [supabase/supabase#2203] [Questions]
Hi, this may be more of a postgres general question than a specific supabase question but I'm having trouble finding a good way to deal with bodies of text that contain line breaks.

My site has a bio element, and the bios tend to be quite long and are often paragraphed. They are being inserted into a table via a standard form input textarea type (the database column is varchar). When they get inserted, the line breaks get retained as per the attached screenshot (at least on the SB UI).

However, when pulling the data and then displaying the bios on the site, I lose the line breaks and it all becomes one giant block of text. Any good solutions for this? Thanks
![Untitled](https://user-images.githubusercontent.com/67643857/124353077-89f11900-dc47-11eb-9d31-e0b0ade8033d.png)

**Comment by burggraf** [comment]
If the line breaks are being stored in the database as `\n` newline characters, you should just replace them before displaying them.  `str.replace(/\n/g, '<br/>')` or something like that.  Hope this helps.
